### PR TITLE
feat: testnet banner

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Some users find their way to the application deployed on testnet (source [forum](https://forum.dfinity.org/t/internet-identity-doubt-in-creating-a-new-ii-anchor/11179/8?u=peterparker)). To make them aware they are accessing a version that has only test purpose, we can display a banner about it.

# Changes

- test purpose banner displayed if environment is "testnet" and not "local development"
- css variable `--header-offset` for layout

# Screenshots

Capture d’écran 2022-03-01 à 08 01 15
Capture d’écran 2022-03-01 à 08 01 38
Capture d’écran 2022-03-01 à 08 01 43

### CODE REVIEW

File: frontend/svelte/src/lib/components/common/Banner.svelte [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  import { i18n } from "../../stores/i18n";
+  import IconClose from "../../icons/IconClose.svelte";
+
+  export let headless: boolean = false;
+
+  let visible: boolean = JSON.parse(
+    localStorage.getItem("nnsdapp-testnet-banner-display") || "true"
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/common/Banner.svelte:8]: maybe move the name of the storage key to a variable

File: frontend/svelte/src/lib/components/common/Banner.svelte [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  const close = () => {
+    visible = false;
+
+    localStorage.setItem("nnsdapp-testnet-banner-display", "false");
+  };
+</script>
+
+<svelte:head>
+  {#if banner && rootStyle}
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/common/Banner.svelte:35]: why not `banner && visible` and make the `rootStyle` always a string?

File: frontend/svelte/src/lib/components/common/Banner.svelte [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  let visible: boolean = JSON.parse(
+    localStorage.getItem("nnsdapp-testnet-banner-display") || "true"
+  );
+
+  const testnet: boolean = process.env.DEPLOY_ENV === "testnet";
+  const localEnv: boolean = JSON.parse(process.env.ROLLUP_WATCH || "false");
+  const banner: boolean = testnet && !localEnv;
+
+  let rootStyle: string | undefined;
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/common/Banner.svelte:15]: why not setting it already at the beginning?

```
const rootStyle = `...`;
```

File: frontend/svelte/src/lib/components/common/HeadlessLayout.svelte [hunk 4]
```diff
@@ -5,10 +5,13 @@
   import Toasts from "../ui/Toasts.svelte";
   import Tooltip from "../ui/Tooltip.svelte";
   import Footer from "./Footer.svelte";
+  import Banner from "./Banner.svelte";
 
   const dispatch = createEventDispatcher();
 </script>
 
+<Banner headless={true} />
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/common/HeadlessLayout.svelte:13]: Does `<Banner headless />` work?

File: frontend/svelte/src/lib/components/common/Banner.svelte [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+<style lang="scss">
+  @use "../../themes/mixins/text";
+
+  div {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+
+    &.headless {
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/common/Banner.svelte:57]: Why is this needed? Could you change something in `HeadlessLayout` to avoid the extra config on the banner?

**Review by lmuntanerDfinity — COMMENTED** [review]
Looks neat! A few comments.

File: frontend/svelte/src/lib/components/common/Banner.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/common/Banner.svelte]: ok

File: frontend/svelte/src/lib/components/common/Banner.svelte (hunk 2 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/common/Banner.svelte]: works too ok

File: frontend/svelte/src/lib/components/common/Banner.svelte (hunk 3 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/common/Banner.svelte]: solve with previous change

File: frontend/svelte/src/lib/components/common/Banner.svelte (hunk 5 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/common/Banner.svelte:55]: That's the most straight forward solution because a css variable cannot be set in the layout itself since the the banner is a child of body. It would need to inject the variable on the body tag which ultimately is kind of uglier as a props.

File: frontend/svelte/src/lib/components/common/HeadlessLayout.svelte (hunk 4 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/common/HeadlessLayout.svelte:13]: Yes that would work too.

File: frontend/svelte/src/lib/components/common/Banner.svelte (hunk 5 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/common/Banner.svelte:55]: Yeah, I meant more changing the HeadlessLayout css so that this banner could also use `position: absolute;` but I see in the `HeadlessLayout` that the position needs to be relative to make the shadow visible. If it doesnot have a straightforward solution, this is extra prop is good.

**Review by lmuntanerDfinity — APPROVED** [review]
